File: Bubble Sort/BubbleSort.cs
using System;
using System.Collections.Generic;

namespace ConsoleApp
{
	public static class Program
	{		
		public static void Main(string[] args)
		{
			// deklarasi variable 
			var rand = new Random(0);
			var arr = new List<int>();

			// isi array dengan random data
			for (var i = 0; i < 10; i++)
			{
				arr.Add(rand.Next() % 100);
			}

			// tampilakan isi array
			Console.WriteLine("Array sebelum sorting:");
			for (var i = 0; i < arr.Count; i++)
			{
				Console.WriteLine(arr[i]);
			}
			Console.WriteLine();

			// memanggil Method BubbleSort()
			arr.BubbleSort();

			// tampilakan isi array setelah di sorting
			Console.WriteLine("Array sesudah sorting:");
			for (var i = 0; i < arr.Count; i++)
			{
				Console.WriteLine(arr[i]);
			}
			Console.WriteLine();
			Console.ReadLine();
		}
		public static void BubbleSort(this List<int> item)
		{
			int temp;
			for (var i = 0; i < item.Count; i++)
			{
				for (var j = i; j < item.Count; j++)
				{
					if (item[i] > (item[j]))
					{
						temp = item[j];
						item[j] = item[i];
						item[i] = temp;
					}
				}
			}
		}
    }
}

File: Bubble Sort/bubble-sort.js
function bubbleSort(array) {
	var temp;

	for (var i = 0; i < array.length; i++) {
		for (var j = i; j < array.length; j++) {
			if (array[i] > array[j]) {
				// swap
				temp = array[i];
				array[i] = array[j];
				array[j] = temp;
			}
		}
	}
}

var array = [];

for (var i = 0; i < 100; i++) {
	// Get random number from 1 to 100
	var randomNumber = Math.floor((Math.random() * 100) + 1);
	array.push(randomNumber);
}

console.log(array);

bubbleSort(array);

console.log(array);
